umh.docs.umh.app/content/en/docs/production-guide/backup_recovery/backup-restore-umh.md

```diff
@@ -0,0 +1,159 @@
+| `OutputPath`         | Output path                                | No      | String  |
+| `ParallelJobs`       | Parallel jobs                              | No      | Int32   |
+| `DaysPerJob`         | Days per job                               | No      | Int32   |
+| `EnableGpgSigning`   | Enable GPG signing                         | No      | Boolean |
+| `GpgSigningKeyId`    | GPG encryption key ID                      | No      | String  |
+| `EnableGpgEncryption`| Enable GPG encryption                      | No      | Boolean |
+| `GpgEncryptionKeyId` | GPG encryption key ID                      | No      | String  |
+| `SkipGpgQuestions`   | Skip GPG questions                         | No      | Boolean |
+{{< /table >}}
+
+
+1. Open a terminal and navigate to the folder containing the script. 
+````powershell
+cd <LOCATION_OF_FOLDER>
+````
+2. Execute the script with your desired parameters.
+
+   ```powershell
+   .\backup.ps1 -IP <IP_OF_THE_SERVER_TO_BACK_UP> -GrafanaToken <YOUR_GRAFANA_API_ADMIN_TOKEN> -KubeconfigPath <PATH_TO_KUBECONFIG> -OutputPath <PATH_TO_LOTS_OF_SPACE>
+   ```
+   {{% notice note %}}
+   If the **OutputPath** is not set, the backup will be stored in the current folder.
+   {{% /notice %}}
+
+3. Wait until the backup is created. This can take a while and your PC needs to stay awake and connected to the server.
+
+   {{% notice warning %}}
+   If the connection is interrupted, there is currently no option to resume the process, therefore you will need to start again.
+   {{% /notice %}}
+
+## Restore
+
+There are four different scripts to restore different components of the United Manufacturing Hub. Each of these scripts
+need to be executed separately. 
+Open a terminal and navigate to the backup folder using:
+````powershell
+cd <LOCATION_OF_FOLDER>
+````
+
+### Restore cluster configuration
+To restore **Helm** execute the **.\restore-helm.ps1** script with the following parameters:
+
+   ```powershell
+   .\restore-helm.ps1 -KubeconfigPath <PATH_TO_KUBECONFIG_OF_THE_NEW_SERVER> -BackupPath <FULL_PATH_TO_BACKUP_FOLDER>
+   ```
+Wait after the Helm restore is done and for all pods to be running again.
+
+### Restore Grafana dashboards
+Execute **.\restore-grafana.ps1** to restore Grafana.
+
+   ````powershell
+   .\restore-grafana.ps1 -FullUrl http://<IP_OF_YOUR_NEW_SERVER>:8080 -Token <YOUR_GRAFANA_API_ADMIN_TOKEN_ON_THE_NEW_SERVER> -BackupPath <FULL_PATH_TO_BACKUP_FOLDER>
+   ````
+
+### Restore Node-RED flows
+
+Execute **.\restore-nodered.ps1** to restore Node-RED.
+
+   ````powershell
+   .\restore-nodered.ps1 -KubeconfigPath <PATH_TO_KUBECONFIG_OF_THE_NEW_SERVER> -BackupPath <FULL_PATH_TO_BACKUP_FOLDER>
+   ````
+
+### Restore the database
+
+Execute **.\restore-timescale.ps1** to restore the database.
+
+   ````powershell
+   .\restore-timescale.ps1 -Ip <IP_OF_YOUR_NEW_SERVER> -BackupPath <FULL_PATH_TO_BACKUP_FOLDER> -PatroniSuperUserPassword <TIMESCALEDB_SUPERUSERPASSWORD>
+   ````
+
```

Comment:
``````markdown
Each component of the United Manufacturing Hub can be restored separately, in
order to allow for more flexibility and to reduce the damage in case of a
failure.

### Cluster configuration

To restore the Kubernetes cluster, execute the `.\restore-helm.ps1` script with
the following parameters:

```powershell
.\restore-helm.ps1 -KubeconfigPath <PATH_TO_KUBECONFIG> -BackupPath <PATH_TO_BACKUP_FOLDER>
```

Verify that the cluster is up and running by opening {{< resource type="lens" name="name" >}}
and checking if the workloads are running.

### Grafana dashboards

To restore the Grafana dashboards, you first need to create a Grafana API Key
for an admin user in the new cluster by following these steps:

1. Open the Grafana UI in your browser and log in with an admin user.
2. Click on the **Settings** icon in the left sidebar and select **API Keys**.
3. Give the API key a name and change its role to **Admin**.
4. Optionally set an expiration date.
5. Click **Add**.
6. Copy the generated API key and save it for later.

Then, on your local machine, execute the `.\restore-grafana.ps1` script
with the following parameters:

   ````powershell
   .\restore-grafana.ps1 -FullUrl http://<IP_OF_THE_SERVER>:8080 -Token <GRAFANA_API_KEY> -BackupPath <PATH_TO_BACKUP_FOLDER>
   ````

### Node-RED flows

To restore the Node-RED flows, execute the `.\restore-nodered.ps1` script with
the following parameters:

   ````powershell
   .\restore-nodered.ps1 -KubeconfigPath <PATH_TO_KUBECONFIG> -BackupPath <PATH_TO_BACKUP_FOLDER>
   ````

### Database

To restore the database, execute the `.\restore-timescale.ps1` script with the
following parameters:

   ````powershell
   .\restore-timescale.ps1 -Ip <IP_OF_THE_SERVER> -BackupPath <PATH_TO_BACKUP_FOLDER> -PatroniSuperUserPassword <DATABASE_PASSWORD>
   ````

``````